Oracle/PCORNetInit.sql

```diff
@@ -0,0 +1,1000 @@
+
+For now, let's count all patients for testing with the KUMC test patients.
+*/
+
+--create or replace view i2b2loyalty_patients as (select patient_num,to_date('01-Jul-2010','dd-mon-rrrr') period_start,to_date('01-Jul-2014','dd-mon-rrrr') period_end from "&&i2b2_data_schema".loyalty_cohort_patient_summary where BITAND(filter_set, 61511) = 61511 and patient_num in (select patient_num from i2b2patient))
+--/
+
+BEGIN
+PMN_DROPSQL('DROP TABLE pcornet_codelist');
+END;
+/
+create table pcornet_codelist(codetype varchar2(20), code varchar2(50))
+/
+
+
+create or replace procedure pcornet_parsecode (codetype in varchar, codestring in varchar) as
+
+tex varchar(2000);
+pos number(9);
+readstate char(1) ;
+nextchar char(1) ;
+val varchar(50);
+
+begin
+
+val:='';
+readstate:='F';
+pos:=0;
+tex := codestring;
+FOR pos IN 1..length(tex)
+LOOP
+--	dbms_output.put_line(val);
+    	nextchar:=substr(tex,pos,1);
+	if nextchar!=',' then
+		if nextchar='''' then
+			if readstate='F' then
+				val:='';
+				readstate:='T';
+			else
+				insert into pcornet_codelist values (codetype,val);
+				val:='';
+				readstate:='F'  ;
+			end if;
+		else
+			if readstate='T' then
+				val:= val || nextchar;
+			end if;
+		end if;
+	end if;
+END LOOP;
+
+end pcornet_parsecode;
+/
+
+
+create or replace procedure pcornet_popcodelist as
+
+codedata varchar(2000);
+onecode varchar(20);
+codetype varchar(20);
+
+cursor getcodesql is
+select 'RACE',c_dimcode from pcornet_demo where c_fullname like '\PCORI\DEMOGRAPHIC\RACE%'
+union
+select 'SEX',c_dimcode from pcornet_demo where c_fullname like '\PCORI\DEMOGRAPHIC\SEX%'
+union
+select 'HISPANIC',c_dimcode from pcornet_demo where c_fullname like '\PCORI\DEMOGRAPHIC\HISPANIC%';
+
+begin
+open getcodesql;
+LOOP 
+	fetch getcodesql into codetype,codedata;
+	EXIT WHEN getcodesql%NOTFOUND ;
+ 	pcornet_parsecode (codetype,codedata );
+end loop;
+
+close getcodesql ;
+end pcornet_popcodelist;
+/
+
+BEGIN
+PMN_DROPSQL('DROP TABLE i2pReport');
+END;
+/
+create table i2pReport (runid number, rundate date, concept varchar(20), sourceval number, destval number, diff number)
+/
+
+BEGIN
+insert into i2preport (runid) values (0);
+pcornet_popcodelist;
+END;
+/
+
+
+--------------------------------------------------------------------------------
+-- DEMOGRAPHIC
+--------------------------------------------------------------------------------
+
+BEGIN
+PMN_DROPSQL('DROP TABLE demographic');
+END;
+/
+CREATE TABLE demographic(
+	PATID varchar(50) NOT NULL,
+	BIRTH_DATE date NULL,
+	BIRTH_TIME varchar(5) NULL,
+	SEX varchar(2) NULL,
+	HISPANIC varchar(2) NULL,
+	BIOBANK_FLAG varchar(1) DEFAULT 'N',
+	RACE varchar(2) NULL,
+	RAW_SEX varchar(50) NULL,
+	RAW_HISPANIC varchar(50) NULL,
+	RAW_RACE varchar(50) NULL
+)
+/
+
+
+--------------------------------------------------------------------------------
+-- ENROLLMENT
+--------------------------------------------------------------------------------
+
+BEGIN
+PMN_DROPSQL('DROP TABLE enrollment');
```

Comment:
That is an update we had talked about and that I would like to make, but time constraints are pushing me to wrap this up for now.  Perhaps we can make a note of it and revisit.